src/components/shared/TokenAmountDisplay.tsx

```diff
@@ -1,16 +1,18 @@
 import { formatNumber } from "@/lib/utils";
 import { BigNumberish } from "ethers";
 import React, { useMemo } from "react";
+import Tenant from "@/lib/tenant/tenant";
+const { token } = Tenant.current();
 
 export default function TokenAmountDisplay({
   amount,
-  decimals,
-  currency,
+  decimals = token.decimals,
+  currency = token.symbol,
   maximumSignificantDigits = 2,
 }: {
   amount: BigNumberish;
-  decimals: number;
-  currency: string;
+  decimals?: number;
+  currency?: string;
   maximumSignificantDigits?: number;
 }) {
   const formattedNumber = useMemo(() => {
```

Comment:
Oh I see. My preference would be to only keep the `amount` in the props and make decimals and symbol instance variables. But this also works.

Reply:
I thought this is what you meant, lmk otherwise. I think this is great since reduces a lot of duplicate code while still giving some flexibility :)